contracts/core/Allo.sol

```diff
@@ -249,44 +258,47 @@ contract Allo is Initializable, Ownable, AccessControl {
         address _token,
         uint256 _amount,
         Metadata memory _metadata,
-        address[] memory _owners
+        address[] memory _managers
     ) internal returns (uint256 poolId) {
         if (!registry.isOwnerOrMemberOfIdentity(_identityId, msg.sender)) {
             revert UNAUTHORIZED();
         }
 
+        // access control
+        bytes32 POOL_MANAGER_ROLE = bytes32(poolId);
+        bytes32 POOL_ADMIN_ROLE = keccak256(abi.encodePacked(poolId, "admin"));
+
         Pool memory pool = Pool({
             identityId: _identityId,
             allocationStrategy: _allocationStrategy,
             distributionStrategy: _distributionStrategy,
-            metadata: _metadata
+            metadata: _metadata,
+            managerRole: POOL_MANAGER_ROLE,
+            adminRole: POOL_ADMIN_ROLE
         });
 
         poolId = ++_poolIndex;
         pools[poolId] = pool;
 
+        // grant admin roles to pool creator
+        _grantRole(POOL_ADMIN_ROLE, msg.sender);
+        // set admin role for POOL_MANAGER_ROLE
+        _setRoleAdmin(POOL_MANAGER_ROLE, POOL_ADMIN_ROLE);
+
         // initialize strategies
         // @dev Initialization is expect to revert when invoked more than once
-        _allocationStrategy.initialize(_identityId, poolId, address(this), _initAllocationData);
-        _distributionStrategy.initialize(_identityId, poolId, address(this), _initDistributionData);
+        _allocationStrategy.initialize(address(this), _identityId, poolId, _initAllocationData);
+        _distributionStrategy.initialize(address(this), _identityId, poolId, _token, _initDistributionData);
 
```

Comment:
@thelostone-mc @codenamejason @zobront 

what do you think about adding some sanity checks here?
I think about:
`_allocationStrategy.identityId == _identityId`, `_allocationStrategy.poolId == _poolId`, `_allocationStrategy.allo == address(this)` etc..

just to ensure, that the used strategies are follwing somehow our standards.

Reply:
Not a bad idea but our interfaces don't expose that data. The only way would be to enforce the `identityId` & `poolId` be exposed somewhere Allo can access on the strategies that are being initialized. Am I thinking about this correctly?

![Screenshot 2023-07-10 at 09 44 22](https://github.com/allo-protocol/allo-v2/assets/9419140/87d4a498-4075-4f48-93bc-098c1b29d306)

Reply:
I like this idea as a sanity check, good call. I would recommend renaming the functions to just the variable names `_allocationStrategy.identityId()` so devs can just set the variables as public instead of adding functions.